Change tooltip appearance ( corrected #6022 )

infograf768 opened the pull request:
See https://github.com/joomla/joomla-cms/pull/6022

I am also taking off an obsolete `text-align: left;` as text-alignment is decided by language.
Also, generatecss.php was not used in a former PR and the change concerning `.nav-collapse.collapse.in` was not included in the rtl.css.

Please test.

infograf768 committed:
Change tooltip appearance ( corrected #6022 )

Hils commented:
@test

Confirm that tooltip originally was white text on black background and, after adding this patch, is now black text on white background.

Thanks for doing this :)

RoterNagel commented:
Yes, the change is there. But I dont't think it is the best solution or the final. It needs more difference between the tooltip and the backend. A shadow or more border. Because the colors from the tooltip are the same then from the backend.

If you are not focused on the screen while the tooltip appears you need a second to recognize the tooltip.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6050">issues.joomla.org/joomla-cms/6050</a>.</sub>

adhocgraFX commented:
works fine, better readable with color change and better formatted with text align left<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6050">issues.joomla.org/joomla-cms/6050</a>.</sub>

rgmears commented:
@RoterNagel 

Apologies that I haven't figured out yet how to correctly change the appropriate files. (Tabs to spaces, etc.)

If you want the border around the tooltip to be darker in template.less change line 959 
from:
+ border:1px solid #aaa;
to:
+ border:1px solid #111;

And if you want the arrow to be darker in variables.less change line 266 
from: 
+@tooltipArrowColor: #ddd; 
to:
+@tooltipArrowColor: #111; <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6050">issues.joomla.org/joomla-cms/6050</a>.</sub>

Yorgoz commented:
@test works!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6050">issues.joomla.org/joomla-cms/6050</a>.</sub>

bertmert commented:
I love it(!!), but found this in article manager > Permissions.


BEFORE:
![tooltip-before](https://cloud.githubusercontent.com/assets/11038612/6659947/da4e293a-cb8f-11e4-9682-95815cecadac.jpg)

AFTER:
![tooltip-after](https://cloud.githubusercontent.com/assets/11038612/6659949/e321bb8a-cb8f-11e4-932d-25a55136e03d.jpg)

(Joomla! 3.4.1-dev Development [ Ember ] 24-February-2015 23:00 GMT )

bertmert commented on administrator/templates/isis/css/template-rtl.css:
```diff
@@ -7805,11 +7805,40 @@ td.nowrap.has-context {
+}
+.tooltip-inner strong {
+	display: block;
+	margin: -6px -8px -12px;
```
See images in my comment. There the tooltip is:

title="Delete New setting for &lt;strong&gt;delete actions&lt;/strong&gt; on this article and the calculated setting based on the parent category and group permissions."

Thus, the words "delete actions" are in a separated line.

bertmert commented on administrator/templates/isis/css/template-rtl.css:
```diff
@@ -7805,11 +7805,40 @@ td.nowrap.has-context {
+	border: 1px solid #aaa;
+}
+.tooltip-inner strong {
+	display: block;
```
See images in my comment. There the tooltip is:

title="Delete New setting for <strong>delete actions</strong> on this article and the calculated setting based on the parent category and group permissions."

Thus, the words "delete actions" are in a separated line.

bertmert commented on administrator/templates/isis/css/template.css:
```diff
@@ -7805,11 +7805,40 @@ td.nowrap.has-context {
+	border: 1px solid #aaa;
+}
+.tooltip-inner strong {
+	display: block;
```
See images in my comment. There the tooltip is:

title="Delete New setting for delete actions on this article and the calculated setting based on the parent category and group permissions."

Thus, the words "delete actions" are in a separated line.

rgmears commented:
@bertmert 
I saw that a few times (also in permissions) but couldn't find an example of it when I looked for it later. 

It seems that the use of the strong element to create a heading in most tooltips was not the best choice. Perhaps those few inline instances could be changed to &lt;b&gt; instead of &lt;strong&gt;.

Either that or change the &lt;strong&gt; headings to &lt;span&gt;. And then &mdash; when it's a heading &mdash; change the less file:
.tooltip-inner strong 
to:
tooltip-inner span

Which of course means a lot of copy/paste in the core tooltip files.

designbengel commented:
I have tested this item :white_check_mark: successfully on 6d1f7e06764fff7468b021f28c2d1b6be6734441<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6050">issues.joomla.org/joomla-cms/6050</a>.</sub>

designbengel commented:
I have tested this item :red_circle: unsuccessfully on 6d1f7e06764fff7468b021f28c2d1b6be6734441<br /><br />Sorry - found that the Backend-loginscreen went white instead of blue background caused by this patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6050">issues.joomla.org/joomla-cms/6050</a>.</sub>

designbengel commented:
The Login Screen went white instead of blue caused by this patch. Can anyone confirm this?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6050">issues.joomla.org/joomla-cms/6050</a>.</sub>

brianteeman commented:
@designbengel is that resolved when you clear the browser cache?

bertmert commented:
Concerning comments
https://github.com/joomla/joomla-cms/pull/6050#issuecomment-81362889
https://github.com/joomla/joomla-cms/pull/6050#issuecomment-81384528

and strong issue

I think the easiest solution to solve would be to change lines
https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/html/html.php#L920-L925

like this (replace strong by span and CSS class)
```
$result = '<span class="tooltip-title">' . $title . '</span>';
...
$result = '<span class="tooltip-title">' . $title . '</span><br />' . $content;
```

then change line
```
.tooltip-inner strong {
```
to
```
.tooltip-inner .tooltip-title {
```
of this PR.

Why: There are too many language placeholders used in tooltips with strong-tags in it and not used as title..

The BR tag between $title and $content was already present.

KingLouis1 commented:
I have tested this item :white_check_mark: successfully on 6d1f7e06764fff7468b021f28c2d1b6be6734441<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6050">issues.joomla.org/joomla-cms/6050</a>.</sub>

bertmert commented:
Sorry, how could you test it successfully? This problem (https://github.com/joomla/joomla-cms/pull/6050#issuecomment-81362889) would break lots of tooltips in ISIS

designbengel commented:
@brianteeman no it stays white in background. ![screen shot 2015-10-24 at 10 54 43](http://issues.joomla.org/uploads/1/27936ce3e6e42b8ca54b64e8f65495ef.png)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6050">issues.joomla.org/joomla-cms/6050</a>.</sub>

bertmert commented:
@designbengel 
@brianteeman 

https://github.com/joomla/joomla-cms/pull/6050#issuecomment-150828726

I compared template.less before and after this patch

This line is removed
```
background-color: @loginBackground;
```

https://github.com/joomla/joomla-cms/blob/staging/administrator/templates/isis/less/template.less#L89

Sorry, couldn't find the patch where this happened

infograf768 committed:
ocrrecting tooltip for <strong> issue.

infograf768 commented:
@bertmert
The PR was far behind as it was made in February, therefore the error you got.

I have now updated it, following your suggestions.
I changed a bit to use font-weight: bold; for the Title and I also changed the border to 2px to make the tooltip  stahd out better.

test for strong can be done in Languages->Overrides ->Edit an override or create a New one in the Text field tooltip.

![screen shot 2015-10-25 at 09 31 25](https://cloud.githubusercontent.com/assets/869724/10714803/063f1a94-7aff-11e5-90d7-63836fac7879.png)

infograf768 commented:
If issues, I will make a new PR

infograf768 commented:
yeah I was sure of that ;)
Folks, you can still test the results but we need new system tests as we move from <strong> to a span.

@Bakual  @zero-24 
Could you help for that as this is out of my scope ;)

bertmert commented:
@infograf768 
Thank you. Now I understand. For me Github and PRs (hierarchy) are still obfuscating.

infograf768 commented:
Please test and comment in https://github.com/joomla/joomla-cms/pull/8150
I am closing this one.
